indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,372 @@
+                _bounds: {_northEast: northEast, _southWest: southWest}
+            }
+        } = evt;
+
+        if (layerType !== 'rectangle') {
+            return;
+        }
+
+        this.setState({
+            areaProps: {bounds: {northEast, southWest}}
+        });
+
+        if (this.featureGroup) {
+            // this is necessary here so that we don't end up with
+            // the initial layer being still visible on the map
+            this.featureGroup.leafletElement.eachLayer((layer) => {
+                const {options: {id}} = layer;
+                if (!id) {
+                    this.featureGroup.leafletElement.removeLayer(layer);
+                }
+            });
+        }
+    };
+
+    editArea = (evt) => {
+        if (!Object.keys(evt.layers._layers).length) {
+            return;
+        }
+
+        const {actions: {updateMapAreas}} = this.props;
+        const editedAreas = [];
+
+        evt.layers.eachLayer(({options: {id}, _bounds}) => {
+            const {_southWest: southWest, _northEast: northEast} = _bounds;
+            editedAreas.push({
+                id,
+                bounds: {southWest, northEast}
+            });
+        });
+
+        updateMapAreas(editedAreas);
+    };
+
+    deleteMapAreas = (evt) => {
+        if (!Object.keys(evt.layers._layers).length) {
+            return;
+        }
+
+        const {actions: {deleteMapAreas}} = this.props;
+        const deletedAreas = [];
+
+        evt.layers.eachLayer(({options: {id}}) => {
+            deletedAreas.push(id);
+        });
+
+        deleteMapAreas(deletedAreas);
+    };
+
+    drawAreas = () => {
+        if (this.map === null) {
+            return null;
+        }
+
+        const {areas} = this.props;
+        return areas.map((area) => {
+            const southWest = [area.top_left_latitude, area.top_left_longitude];
+            const northEast = [area.bottom_right_latitude, area.bottom_right_longitude];
+            const latLngBounds = new Leaflet.LatLngBounds([southWest, northEast]);
+            const center = this.map.leafletElement.latLngToContainerPoint(latLngBounds.getCenter());
+            const bottomRight = this.map.leafletElement.latLngToContainerPoint(latLngBounds.getSouthEast());
+            const onAreaClick = () => {
+                const {suppressClick} = this.state;
+                if (suppressClick) {
+                    return;
+                }
+
+                this.setState({
+                    areaProps: {
+                        bounds: {
+                            southWest: latLngBounds.getSouthWest(),
+                            northEast: latLngBounds.getNorthEast()
+                        },
+                        id: area.id,
+                        isDefault: area.is_default,
+                        name: area.name
+                    }
+                });
+            };
+
+            return (
+                <Rectangle key={`${area.name}-${center.x}-${center.y}`}
```

Comment:
why not the ID?

Reply:
Actually, not sure why I didn't use id here

Reply:
Ah, now I remember. In case when I use `id` as the value for the `key` prop, rectangles (areas) are not rerendered when e.g. editing the size/coordinates (which for example affects position of labels).